### PR TITLE
Feat/add my adoptions list screen

### PR DESCRIPTION
**Pull request by Isaac-alencar**
closes: #194 

## Overview
Add new screen with all adoptions created by a user. Since we don't have an endpoint to return such list, we mocked the data within the `AdoptionsScreen`.

## UI
  <div style="display:grid;width="100%;gap:2rem;"">
  link under user profile
  my adoptions list screen
</div>

### CODE REVIEW

File: apps/mobile/src/i18n/locales/pt-BR.json
Comment (belgamo) [inline apps/mobile/src/i18n/locales/pt-BR.json]:
```suggestion
    "TITLE":"Meus Pets",
```

File: apps/mobile/src/navigation/main-navigator.tsx
Comment (belgamo) [inline apps/mobile/src/navigation/main-navigator.tsx]: Let's name it `MyAdoptions` cause `Adoptions` is a very generic term.

File: apps/mobile/src/screens/adoptions/components/adoptions-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoptions-list.tsx]: Can you build a more meaningful string to prevent it from conflicting with other lists? Something like: 

`keyExtractor={({ id }) => `my-adoption-item-${id}`}`

File: apps/mobile/src/routes.ts
Comment (belgamo) [inline apps/mobile/src/routes.ts]:
```suggestion
  MyAdoptions: 'MyAdoptions',
```

File: apps/mobile/src/i18n/locales/pt-BR.json
Comment (belgamo) [inline apps/mobile/src/i18n/locales/pt-BR.json]:
```suggestion
    "SUB_TITLE":"Confira os pets que você registrou para adoção.",
```

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]:
```suggestion
const MyPetsScreen = () => {
```

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]:
```suggestion
        <MyPetsList adoptions={DATA} />
```

File: apps/mobile/src/screens/adoptions/components/adoption-card.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoption-card.tsx]:
```suggestion
export const MyPetCard = ({ name, type }: AdoptionType) => {
```

Rename the file as well to `my-pet-card.tsx`

File: apps/mobile/src/screens/adoptions/components/adoptions-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoptions-list.tsx]: Please move this to the location file and translate to portuguese.

File: apps/mobile/src/screens/adoptions/components/adoptions-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoptions-list.tsx]: Fair enough, you can get rid of this component.

File: apps/mobile/src/screens/adoptions/components/header.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/header.tsx]:
```suggestion
          {t('MY_PETS_SCREEN.TITLE')}
```

File: apps/mobile/src/i18n/locales/pt-BR.json
Comment (belgamo) [inline apps/mobile/src/i18n/locales/pt-BR.json]:
```suggestion
  "MY_PETS_SCREEN": {
```

File: apps/mobile/src/screens/adoptions/components/adoption-card.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoption-card.tsx:1]: Please rename the `components` folder to `compose`

File: apps/mobile/src/screens/adoptions/components/adoptions-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/components/adoptions-list.tsx]:
```suggestion
export const MyPetsList = ({ adoptions }: AdoptionsListProps) => {
```

Rename the file as well to `my-pets-list.tsx`

**Review by belgamo — CHANGES_REQUESTED** [review]
Overall it looks good, just a few comments to reword. Please add tests to your components.

I'm gonna take care of the endpoint so let's hold off this PR until then.

File: apps/mobile/__tests__/screens/adoptions/MyPetsList.test.tsx
Comment (belgamo) [inline apps/mobile/__tests__/screens/adoptions/MyPetsList.test.tsx]:
```suggestion
describe('<MyPetsList />', () => {
  describe('when user has not put pets for adoptions yet', () => {
    it('renders an empty message', () => {
      renderWithProviders(<MyPetsList adoptions={[]} />);

      const emptyListMsg = screen.getByText('Você ainda não registou nenhum pet para adoção');

      expect(emptyListMsg).toBeOnTheScreen();
    });
  });

  describe('when user has pets for adoptions', () => {
    it('renders registered pets in a list', () => {
      renderWithProviders(<MyPetsList adoptions={DATA} />);

      for (const pet of DATA) {
        const petName = pet.name.toUpperCase();
        expect(screen.getByText(petName)).toBeOnTheScreen();
      }
    });
  });
});
```

WDYT? nested describes blocks are optional if you don't like them.

File: apps/mobile/src/components/topbar/topbar.tsx
Comment (belgamo) [inline apps/mobile/src/components/topbar/topbar.tsx]: please use the hook `useLocale` instead

File: apps/mobile/__tests__/screens/adoptions/MyPetsList.test.tsx
Comment (belgamo) [inline apps/mobile/__tests__/screens/adoptions/MyPetsList.test.tsx:1]: this file should be under `screens/adoptions/compose` close to the actual component. Please, rename this to `my-pets-list.test.tsx`

File: apps/mobile/src/screens/adoptions/compose/adoptions-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/adoptions-list.tsx:1]: please rename this to `my-pets-list.tsx`

File: apps/mobile/src/components/topbar/topbar.tsx
Comment (belgamo) [inline apps/mobile/src/components/topbar/topbar.tsx]: this component is supposed to be reused across the app screens meaning that you can't have a fixed title here. all this code should be moved to `profile.screen.tsx` and you should use a list instead. ping me on discord if you have any questions.

File: apps/mobile/src/screens/adoptions/compose/header.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/header.tsx:1]: you can delete this since you should be using the `topbar` component instead.

**Review by belgamo — CHANGES_REQUESTED** [review]

File: apps/mobile/src/components/topbar/topbar.tsx
Comment (belgamo) [inline apps/mobile/src/components/topbar/topbar.tsx]: please remove this import

File: apps/mobile/src/components/topbar/topbar.tsx
Comment (belgamo) [inline apps/mobile/src/components/topbar/topbar.tsx]: please remove `navigate` since it's not being used

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx]: please remove the box import

File: apps/mobile/src/components/topbar/topbar.tsx
Comment (belgamo) [inline apps/mobile/src/components/topbar/topbar.tsx]:
```suggestion
    <Box display="flex" flexDirection="row">
```

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]: there is a typescript error here

File: apps/mobile/src/screens/adoptions/compose/header.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/header.tsx]: you cand reuse the existing topbar component instead

File: apps/mobile/src/screens/adoptions/compose/my-pets-list.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/my-pets-list.tsx]: https://github.com/animavita/animavita/assets/19699724/d7fde013-2c5c-49a5-99da-2f2d68c54299

there is a bug when trying to get to the last card in the list

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx]: I think a `Pressable` component is more convenient to be used here. The text should be also left aligned

**Review by belgamo — CHANGES_REQUESTED** [review]

**Comment by Isaac-alencar** [comment]
As we talked earlier, I made too many commit messages, I can squash them into a single one that highlight what was done, before it got merged. What do you think?

File: apps/mobile/src/screens/adoptions/compose/header.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/header.tsx:1]: please remove this file

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx]:
```suggestion
    name: 'my-pets',
```

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx]:
```suggestion
const PROFILE_ROUTES = [
```

File: apps/mobile/src/screens/adoptions/compose/header.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/compose/header.tsx]: move this code to `adoptions.screen.tsx`

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]:
```suggestion
const MyPetsScreen = () => {
```

everything that says "my adoptions" should be renamed to "my pets"

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]: please remove this import

File: apps/mobile/src/screens/adoptions/adoptions.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/adoptions/adoptions.screen.tsx]:
```suggestion
    <SafeArea>
      <Delimiter flex="1">
        <Topbar />
        <Box mt="4">
          <MyPetsList adoptions={DATA as Adoption[]} />
        </Box>
      </Delimiter>
    </SafeArea>
```

use the `Delimiter` component

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx]:
```suggestion
              mb={2}
```

you should be trying to fix the card alignment here. actually the one that seems misaligned is the arrow in the `TopBar` component, so don't worry about it.

File: apps/mobile/src/screens/profile/profile.screen.tsx
Comment (belgamo) [inline apps/mobile/src/screens/profile/profile.screen.tsx:22]:
```suggestion
        <FlatList
          mt="4"
```

**Review by belgamo — CHANGES_REQUESTED** [review]

**Comment by belgamo** [comment]
> As we talked earlier, I made too many commit messages, I can squash them into a single one that highlight what was done, before it got merged. What do you think?

don't worry about it in this PR, you can use the squash strategy for your next contributions. thanks!

**Review by belgamo — APPROVED** [review]

**Pull request merged**
